modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/outgoing/OutgoingSnapshotsManager.java

```diff
@@ -103,17 +103,17 @@ public void stop() throws Exception {
     }
 
     /**
-     * Registers an outgoing snapshot in the manager.
+     * Starts an outgoing snapshot and registers it in the manager. This is the point where snapshot is 'taken',
+     * that is, the immutable scope of the snapshot (what MV data and what TX data belongs to it) is cut.
      *
      * @param snapshotId       Snapshot id.
      * @param outgoingSnapshot Outgoing snapshot.
      */
-    @Override
-    public void registerOutgoingSnapshot(UUID snapshotId, OutgoingSnapshot outgoingSnapshot) {
+    void startOutgoingSnapshot(UUID snapshotId, OutgoingSnapshot outgoingSnapshot) {
         snapshots.put(snapshotId, outgoingSnapshot);
 
         PartitionSnapshotsImpl partitionSnapshots = getPartitionSnapshots(outgoingSnapshot.partitionKey());
```

Comment:
Original code (written by you) was using concrete type. I added an interface (for registering/unregistering), it seemed useful because it was both used by prod code and mocked in tests. But now, after refactorings, it turned out that the interface is not needed *here*, so I removed it.

If you still like the interface, let me know, I will restore it.

Reply:
I don't like using specific implementation when it's not required. It's like writing `ArrayList<?> list = new ArrayList<>()`, just unnecessary, unless you call something like `trimToSize`, which is a very specific operation

Reply:
Oh, sorry, I misunderstood you. Yes, we need access to `PartitionSnapshotsImpl` internals here